client/src/offline-settings/mdn-worker.tsx

```diff
@@ -226,71 +134,17 @@ export class MDNWorker {
       settingsData.autoUpdates === true &&
       current.autoUpdates === false
     ) {
-      await this.autoUpdate();
+      this.autoUpdate();
     }
 
     const settings = { ...current, ...settingsData };
     window.localStorage.setItem("MDNSettings", JSON.stringify(settings));
     this.settings = settings;
     return settings;
   }
-  clear() {
-    this.updateStatus.state = STATE.clearing;
+  async clear() {
     this.controller()?.postMessage({ type: "clear" });
   }
-
-  async setStatus(updateStatus: UpdateStatus) {
-    if (typeof updateStatus.progress !== "undefined") {
-      this.updateStatus.progress = updateStatus.progress;
-    }
-    if (typeof updateStatus.currentDate !== "undefined") {
-      this.updateStatus.currentDate = updateStatus.currentDate;
-    }
-    if (typeof updateStatus.currentVersion !== "undefined") {
-      this.updateStatus.currentVersion = updateStatus.currentVersion;
-    }
-    await this.setOfflineSettings({
-      currentDate: this.updateStatus.currentDate,
-      currentVersion: this.updateStatus.currentVersion || null,
-    });
-    if (updateStatus.state) {
-      if (
-        updateStatus.state === STATE.init &&
-        this.updateStatus.state !== STATE.init
-      ) {
-        this.updating = false;
-        this.updateStatus.updateVersion = null;
-        this.updateStatus.updateDate = null;
-        this.updateStatus.state = updateStatus.state;
-        await this.getUpdate();
-      } else {
-        if (
-          [STATE.clearing, STATE.downloading, STATE.unpacking].includes(
-            updateStatus.state
-          )
-        ) {
-          this.updating = true;
-        } else {
-          this.updating = false;
-        }
-
-        if (this.updating && !this.keepAlive) {
-          this.keepAlive = setInterval(() => {
-            this.controller()?.postMessage({ type: "keepalive" });
-            if (!this.updating && this.keepAlive) {
-              clearInterval(this.keepAlive);
-              this.keepAlive = null;
-            }
-          }, 10000);
-        }
```

Comment:
@fiji-flo I wonder whether we still need a keepAlive mechanism like this, and if so, how best to integrate it now that the client is essentially just listening during the update?

Reply:
Just to check:

1. Is this keepAlive only necessary during download/unpack/clear phase, or all the time?
2. Should we then also prevent the user from navigating away during those phases?

Reply:
Fixed in 3c8c1813c (keepalive) and 1eaeeed6c (warn when navigating away).